### PR TITLE
Add : Hardware manager thread and sensor cache

### PR DESCRIPTION
**Pull request by RaulTrombin**

### CODE REVIEW

File: src/hardware_manager.rs [hunk 1]
```diff
@@ -1,24 +1,14 @@
 use lazy_static::lazy_static;
 use std::convert::From;
-use std::sync::Mutex;
-
-struct NavigationManager {
+use std::sync::{Arc, Mutex, RwLock};
+use std::thread;
+pub struct NavigationManager {
```
Comment (joaoantoniocardoso) [inline src/hardware_manager.rs:5]: Does this need to be pub?

File: src/hardware_manager.rs [hunk 2]
```diff
@@ -1,24 +1,14 @@
...
-        if NAVIGATOR.lock().unwrap().is_none() {
-            *NAVIGATOR.lock().unwrap() = Some(NavigationManager {
-                navigator: navigator_rs::Navigator::new(),
-            });
-        }
-        &NAVIGATOR
-    }
+#[derive(Debug, Clone, Default)]
+pub struct Data {
+    pub state: navigator_rs::SensorData,
```
Comment (joaoantoniocardoso) [inline src/hardware_manager.rs:11]: Do these need to be pub?

Comment (joaoantoniocardoso) [reply src/hardware_manager.rs:11]: I see it is being used as the return type of Data::read, but Data::read might not need to be exposed

Comment (RaulTrombin) [reply src/hardware_manager.rs:11]: it needn't, the functions use it from same module and return derived values, ok

**Pull request merged**